### DISCUSSION TITLE
About `AsyncLoading` loading message

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#51] [General]
As `AsyncError` has its own data (error and stack trace), would be helpful to have an optional `Q extends Object` field inside `AsyncLoading`, typically a `String`. This would be helpful to show specific loading when warming up capsules (loading spinner + text info).
This may require to add an extra generic type to `AsyncLoading` (`<Q extends Object>`).

What do you think?

**Comment by busslina** [comment]
PR: https://github.com/GregoryConrad/rearch-dart/pull/52

**Comment by GregoryConrad** [comment]
I would do something like the following:

```dart
[
  ('logger', loggerCapsule),
  ('cache', cacheCapsule),
  // ...
].toInfoWarmUpWidget(
  // similar args as toWarmUpWidget, but also has loading/error builder that has info in addition to AsyncValue
);
```

I would gladly accept a PR for such a method. The first element of each tuple can be a generic T, but I imagine most users will use strings.

or maybe name it toWarmUpWidgetWithInfo. Up to you!

**Reply by busslina** [reply]
Cool solution.

I have a mess in my brain and I can not  think efficiently (Too much work migrating from Riverpod to ReArch ;)).
